doc/tutorial/extending_theano.txt

```diff
@@ -297,6 +297,78 @@ You can try it as follows:
     print out
 
 
+Example for properties of a Op
+==========
+
+We can modify the previous piece of code in order to demonstrate
+the usage of the :attr:`__props__` attribute.
+
+We create an Op that takes a variable ``x`` and returns ``a*x+b``.
+We want to say that two such ops are equal when their values of ``a``
+and ``b`` are equal.
+
+.. code-block:: python
+
+    import theano
+
+    class AXPBOp(theano.Op):
+        """
+        This creates an Op that takes x to a*x+b.
+        """
+        __props__ = ("a", "b")
+
+        def __init__(self, a, b):
+            self.a = a
+            self.b = b
+            super(AXPBOp, self).__init__()
+
+        def make_node(self, x):
+            # check that the theano version has support for __props__
+            assert hasattr(self, '__props__')
+            x = theano.tensor.as_tensor_variable(x)
+            return theano.Apply(self, [x], [x.type()])
+
+        def perform(self, node, inputs, output_storage):
+            x = inputs[0]
+            z = output_storage[0]
+            z[0] = self.a * x + self.b
+
+        def infer_shape(self, node, i0_shapes):
+            return i0_shapes
+
+        def grad(self, inputs, output_grads):
+            return [a * output_grads[0] + b]
+
+
+The use of :attr:`__props__` saves
+the user the trouble of implementing :func:`__eq__` and :func:`__hash__` manually.
```

Comment:
It also generates a default `__str__` method that prints the attribute names and their values.